File: material/strong/filt.java
import java.util.Arrays;

class Filt {
	public static void main(String[] args) {
		int[] d = {3, -1, 4, -2, 5, -3, 6, -4};
		int[] d2 = filt(d);
		for(int v : d2) {
			System.out.println(v);
		}
	}

	public static int[] filt(int[] arr) {
		return Arrays.stream(arr).filter(x -> x < 0).toArray();
	}
}

File: material/strong/filt.rs
extern crate num;
use num::Num;

fn main() {
    let d = [3, -1, 4, -2, 5, -3, 6, -4];
    let d2 = filt(&d);
    for i in d2 {
        println!("{}", i);
    }
}

fn filt<T: Num + PartialOrd + Copy>(arr: &[T]) -> Vec<T> {
    arr.into_iter().filter(|&&i| i < T::zero()).cloned().collect()
}
